proto/ibc/core/channel/v1/channel.proto

```diff
@@ -90,6 +90,21 @@ enum Order {
   ORDER_ORDERED = 2 [(gogoproto.enumvalue_customname) = "ORDERED"];
 }
 
+// IBCVersion defines the version that the IBC packet intends to use.
+// This allows asynchronous upgrades over time as unsupported IBC Versions
+// will simply be rejected by the receiver and timed out on sender.
+enum IBCVersion {
+  option (gogoproto.goproto_enum_prefix) = false;
+
+  // zero-value for IBC version.
+  // This will be treated as a classic packet
+  IBC_VERSION_UNSPECIFIED = 0;
+  // IBC version 1 implements the Classic protocol
+  IBC_VERSION_CLASSIC = 1;
+  // IBC version 2 implements the Eureka protocol
+  IBC_VERSION_EUREKA = 2;
```

Comment:
looked over this with @chatton since it overlaps with our work. He made a good point that Eureka name is leaking into the codebase. I think long term, referring to this as v1 and v2 packet commitment structures would be more intuitive. Otherwise you keep adding codenames for every version (which I find to be very confusing)

Reply:
Agreed, switched to v2 for Eureka and v1 for classic. With this context, I think it still makes sense to leave the 0 enum as UNSPECIFIED but treat it as a v1 packet. What do you think?

I prefer that to zero indexing the IBC_Version_1 enum